premake5.lua

```diff
@@ -84,6 +84,40 @@
 	solution "Premake5"
 		configurations { "Release", "Debug" }
 		location ( _OPTIONS["to"] )
+		
+		configuration "Debug"
+			defines     "_DEBUG"
+			flags       { "Symbols" }
+
+		configuration "Release"
+			defines     "NDEBUG"
+			flags       { "OptimizeSize" }
+
+		configuration "vs*"
+			defines     { "_CRT_SECURE_NO_WARNINGS" }
+
+		configuration "vs2005"
+			defines   {"_CRT_SECURE_NO_DEPRECATE" }
+			
+		configuration "cygwin"
+			--gccprefix "i686-pc-cygwin-"
+			defines   { "LUA_USE_POSIX" }
+
+		configuration "linux or bsd or hurd"
+			defines     { "LUA_USE_POSIX", "LUA_USE_DLOPEN" }
+
+		configuration "macosx"
+			defines     { "LUA_USE_MACOSX" }
+
+		configuration { "macosx", "gmake" }
+			toolset "clang"
+			buildoptions { "-mmacosx-version-min=10.4" }
+			linkoptions  { "-mmacosx-version-min=10.4" }
+
+		configuration "aix"
+			defines     { "LUA_USE_POSIX", "LUA_USE_DLOPEN" }
+
+		configuration {}
```

Comment:
Agreed—this is a reasonable change and could have been accepted easily, now it is going to get all tangled up in these other changes.